src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php

```diff
@@ -344,22 +357,30 @@ protected function findAttributeOr404($id)
      * @param ProductInterface $product
      * @param array            $data
      */
-    protected function updateProduct(ProductInterface $product, array $data)
+    protected function updateProduct($product, array $data)
     {
         $values = $this->productValueConverter->convert($data['values']);
 
         $values = $this->localizedConverter->convertToDefaultFormats($values, [
             'locale' => $this->userContext->getUiLocale()->getCode()
         ]);
 
-        $dataFiltered = $this->emptyValuesFilter->filter($product, ['values' => $values]);
-
-        if (!empty($dataFiltered)) {
-            $data = array_replace($data, $dataFiltered);
-        } else {
-            $data['values'] = [];
+//        $dataFiltered = $this->emptyValuesFilter->filter($product, ['values' => $values]);
+//
+//        if (!empty($dataFiltered)) {
+//            $data = array_replace($data, $dataFiltered);
+//        } else {
+//            $data['values'] = [];
+//        }
+
+        $data['values'] = $values;
+        foreach ($data['values'] as $k => $value) {
+            $data['values'][$k] = current($data['values'][$k]);
         }
 
-        $this->productUpdater->update($product, $data);
+        unset($data['identifier']);
+        unset($data['associations']);
+//        $this->productUpdater->update($product, $data);
+        $this->productModelUpdater->update($product, $values);
```

Comment:
I think that it is not a good to have generic updater. `Product` and `ProductModel` are 2 differents objects.